lib/elixir/lib/range.ex

```diff
@@ -461,6 +461,62 @@ defmodule Range do
     end
   end
 
+  @doc """
+  Checks if all elements of range2 are members of range1.
+
+      iex> Range.subset?(1..10, 2..5)
+      true
+      iex> Range.subset?(1..10, 11..12)
+      false
+      iex> Range.subset?(8..20, 1..12)
+      false
+
+  Steps are also considered when checking if the second range is a subset of the first:
+
+      iex> Range.subset?(1..10, 2..4//2)
+      true
+      iex> Range.subset?(2..8//2, 4..8//4)
+      true
+      iex> Range.subset?(2..8//2, 3..7//4)
+      false
+      iex> Range.subset?(2..11//3, 5..11//6)
+      true
+      iex> Range.subset?(11..2//-3, 5..11//6)
+      true
+      iex> Range.subset?(1..10//2, 2..8//3)
+      false
+      iex> Range.subset?(0..10//5, 0..4//2)
+      false
+      iex> Range.subset?(0..10, 1..3//10)
+  """
+  @spec subset?(t, t) :: boolean
+  def subset?(first1..last1//step1 = range1, first2..last2//step2 = range2) do
+    {first1, last1, step1} = normalize(first1, last1, step1)
+    {first2, last2, step2} = normalize(first2, last2, step2)
+    size1 = size(range1)
+    size2 = size(range2)
+
+    cond do
+      size1 == 0 and size2 == 0 ->
+        true
```

Comment:
I thought about adding this function and that's the part I am not sure I agree with.

I think saying two ranges of any size without points are subsets is correct but not practically useful. I would  prefer to have `contains?` and say it checks its bounds are within each other and all points in one belong to the other.

Reply:
yes, it's quite pedantic. after all, the only ranges that are "empty" are ascending (first < last) ranges with a negative step part, right? I'll change the function accordingly.

Reply:
I will reintroduce this check now that there is a contains_bounds? function.